### PR TITLE
Decompile unk_02022450.s

### PR DESCRIPTION
**Pull request by SNBeast**

### CODE REVIEW

File: arm9/src/unk_02022450.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+static const char string_saying_rom[] = "rom";
+
+THUMB_FUNC void FUN_02022450 () {
+    if (!FS_IsAvailable()) {
+        OS_Terminate();
+    }
+    else {
+        u32 *r1 = (u32*)HW_MAIN_MEM_SHARED;
+        if (!*(u32*)0x027FF00C) {
```
Comment (red031000) [inline arm9/src/unk_02022450.c:17]: a constant can probably be created for this, it seems to be some sort of protection value located in shared work, it's equal to HW_MAIN_MEM_SHARED[0xC]

File: arm9/src/unk_02022450.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            MI_CpuCopy8((u8*)HW_ROM_HEADER_BUF, (u8*)r1, 22 * 16);
+            MI_CpuCopy8((u8*)HW_ROM_HEADER_BUF, (u8*)HW_CARD_ROM_HEADER, 22 * 16);
+            *(u32*)0x027FF00C = 0x4A414441;
+        }
+        FSArchive * const r0 = FS_FindArchive(string_saying_rom, 3);
+        r0->fat = r1[18];
+        r0->fat_size = r1[19];
+        r0->fnt = r1[16];
+        r0->fnt_size = r1[17];
+        if (*(u32*)0x027FF00C != 0x4A414441 || *(u16*)0x027FF010 != 0x3130) {
```
Comment (red031000) [inline arm9/src/unk_02022450.c:28]: another constant here, HW_MAIN_MEM_SHARED[0x10]

**Review by red031000 — CHANGES_REQUESTED** [review]
spacing in these files are also really inconsistent, it should be four spaces, not tabs

File: arm9/src/unk_02022450.c (hunk 2 above)
Comment (SNBeast) [reply arm9/src/unk_02022450.c:28]: Or a buffer for the ROM header, more likely.

**Review by red031000 — CHANGES_REQUESTED** [review]
why the intendation changes to the lsf and module 83? they were fine as it, the problem was that there was two different types of indentation in a single file, please revert them back to how they were

**Comment by SNBeast** [comment]
Apparently the lsf was also inconsistently spaced. And it was never an issue.

**Review by red031000 — APPROVED** [review]

**Comment by red031000** [comment]
can you resolve the merge conflicts? then I will merge this

**Review by red031000 — APPROVED** [review]

**Pull request merged**